presto-hive/src/main/java/com/facebook/presto/hive/avro/PrestoAvroSerDe.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.avro;
+
+import org.apache.avro.Schema;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.serde2.avro.AvroSerDe;
+import org.apache.hadoop.hive.serde2.avro.AvroSerdeException;
+import org.apache.hadoop.hive.serde2.avro.AvroSerdeUtils;
+
+import java.io.IOException;
+import java.util.Properties;
+
+public class PrestoAvroSerDe
+        extends AvroSerDe
+{
+    @Override
+    public Schema determineSchemaOrReturnErrorSchema(Configuration conf, Properties props)
+    {
+        try {
+            return AvroSerdeUtils.determineSchemaOrThrowException(conf, props);
+        }
+        catch (IOException | AvroSerdeException e) {
+            throw new RuntimeException(e);
```

Comment:
Just curious, why does the existing version get suppressed?

Reply:
From traces in Hive jira, i think this is because SerDe is initialized within Metastore during ops like ALTER TABLE. When schema file is gone, throwing directly from Serde.init would render the table unmodifiable and unfixable.... -- but this is just my hypothesis, i didn't test that

Reply:
I added some explanation in the code.